h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderHelper.java

```diff
@@ -82,6 +73,90 @@ static double computePriorMean(Frame encodings, int targetClass) {
+      return addInteractionColumn(fr, colGroup, interactionDomain);
+    } else { // skip interaction if one col is missing  (we could also replace missing columns with a col of NAs, but we don't do this today when a simple cat column is missing)
+      return -1;
+    }
+  }
+
+  private static int addInteractionColumn(Frame fr, String[] interactingColumns, String[] interactionDomain) {
+    String interactionColName = String.join("~", interactingColumns);  // any limit to col name length?
+    int[] cols = Arrays.stream(interactingColumns).mapToInt(fr::find).toArray();
+    Vec interactionCol = createInteractionColumn(fr, cols, interactionDomain);
+    fr.add(interactionColName, interactionCol);
+    return fr.numCols()-1;
+  }
+  
+  static Vec createInteractionColumn(Frame fr, int[] interactingColumnsIdx, String[] interactionDomain) {
+    String[][] interactingDomains = new String[interactingColumnsIdx.length][];
+    Vec[] interactingVecs = new Vec[interactingColumnsIdx.length];
+    for (int i=0; i<interactingColumnsIdx.length; i++) {
+      Vec vec = fr.vec(interactingColumnsIdx[i]);
+      interactingVecs[i] = vec;
+      interactingDomains[i] = vec.domain();
+    }
+    final InteractionsEncoder encoder = new InteractionsEncoder(interactingDomains, true);
+    byte interactionType = interactionDomain == null ? Vec.T_NUM : Vec.T_CAT;
+    Vec interactionCol = new CreateInteractionTask(encoder, interactionDomain)
+            .doAll(new byte[] {interactionType}, interactingVecs)
+            .outputFrame(null, null, new String[][]{interactionDomain})
+            .lastVec();
+    if (interactionType != Vec.T_CAT)
+      interactionCol = VecUtils.toCategoricalVec(interactionCol); // the domain is obtained from CollectDoubleDomain, so it is sorted by numerical value, and then converted to String
+    return interactionCol;
+  }
+
+  private static class CreateInteractionTask extends MRTask<CreateInteractionTask> {
+    final InteractionsEncoder _encoder;
+    final long[] _interactionDomain; // sorted by construction (see createInteractionColumn), or null
+    
+    public CreateInteractionTask(InteractionsEncoder encoder,  String[] interactionDomain) {
+      _encoder = encoder;
+      _interactionDomain = interactionDomain == null ? null : Arrays.stream(interactionDomain).mapToLong(Long::parseLong).toArray();
+    }
+
+    @Override
+    public void map(Chunk[] cs, NewChunk nc) {
+      for (int row=0; row < cs[0].len(); row++) {
+        int[] interactingValues = new int[cs.length];
+        for (int i=0; i<cs.length; i++) {
+          interactingValues[i] = cs[i].isNA(row) ? -1 : (int)cs[i].at8(row);
+        }
+        long val = _encoder.encode(interactingValues);
+        if (val < 0) {
+          nc.addNA();
+        } else if (_interactionDomain == null){
+          nc.addNum(val);
+        } else {
+          int catVal = Arrays.binarySearch(_interactionDomain, val);
```

Comment:
Could speed this up by creating a map of sorted indices to the actual values. No need to binary search afterwards, just access the value quickly.

Reply:
true, should mainly be useful for very large interaction domains: I'll try it.